src/main/scala/is/hail/io/vcf/LoadVCF.scala

```diff
@@ -850,67 +887,117 @@ object LoadVCF {
           if (s1 != s2) {
             fatal(
               s"""invalid sample ids: expected sample ids to be identical for all inputs. Found different sample ids at position $i.
-               |    ${ files(0) }: $s1
-               |    $file: $s2""".stripMargin)
+                 |    ${ inputs(0) }: $s1
+                 |    $file: $s2""".stripMargin)
           }
         }
 
         if (hd1.genotypeSignature != hd.genotypeSignature)
           fatal(
             s"""invalid genotype signature: expected signatures to be identical for all inputs.
-             |   ${ files(0) }: ${ hd1.genotypeSignature.toString }
-             |   $file: ${ hd.genotypeSignature.toString }""".stripMargin)
+               |   ${ inputs(0) }: ${ hd1.genotypeSignature.toString }
+               |   $file: ${ hd.genotypeSignature.toString }""".stripMargin)
 
         if (hd1.vaSignature != hd.vaSignature)
           fatal(
             s"""invalid variant annotation signature: expected signatures to be identical for all inputs.
-             |   ${ files(0) }: ${ hd1.vaSignature.toString }
-             |   $file: ${ hd.vaSignature.toString }""".stripMargin)
+               |   ${ inputs(0) }: ${ hd1.vaSignature.toString }
+               |   $file: ${ hd.vaSignature.toString }""".stripMargin)
       }
     } else {
       warn("Loading user-provided header file. The number of samples, sample IDs, variant annotation schema and genotype schema were not checked for agreement with input data.")
     }
 
-    val VCFHeaderInfo(sampleIdsHeader, infoSignature, vaSignature, genotypeSignature, _, _, _, infoFlagFieldNames) = header1
+    val VCFHeaderInfo(sampleIDs, infoSignature, vaSignature, genotypeSignature, _, _, _, infoFlagFieldNames) = header1
 
-    val sampleIds: Array[String] = if (dropSamples) Array.empty else sampleIdsHeader
-    val localNSamples = sampleIds.length
-
-    LoadVCF.warnDuplicates(sampleIds)
+    LoadVCF.warnDuplicates(sampleIDs)
```

Comment:
Suggestion:
```
if (!dropSamples)
  LoadVCF.warnDuplicates(sampleIDs)
```

Reply:
I don't think dropSamples should be exposed to the user in importvcf -- they should just filter all cols after import. To this end, I think this parameter should be just passed into the MatrixRead IR so that we can remove it more easily.